src/main/java/kg/jarkyn/cobspecserver/Runner.java

```diff
@@ -0,0 +1,32 @@
+package kg.jarkyn.cobspecserver;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public class Runner {
+    public static void main(String[] args) {
+        server().run();
+    }
+
+    private static Server server() {
+        return new Server(listener(), controller());
+    }
+
+    private static ListenerSocket listener() {
+        return new ListenerSocket(getServerSocket(), true);
```

Comment:
Or even better, add different constructors to ListenerSocket instead of passing a boolean flag.